src/main/java/uk/gov/pay/directdebit/app/config/DirectDebitModule.java

```diff
@@ -153,4 +154,9 @@ public GoCardlessEventDao provideGoCardlessEventDao() {
         return jdbi.onDemand(GoCardlessEventDao.class);
     }
 
+    @Provides
+    @Singleton
+    public PaymentViewDao providePaymentViewDao() {
+        return new PaymentViewDao(jdbi);
```

Comment:
this should be `jdbi.onDemand(PaymentViewDao.class)` as all the other examples - ondemand works well with dropwizard as it opens and closes the connection to the db automatically. If we decide not to use ondemand (because of transactions and so on) and attach it to an handle, it's something we can think of later when we play the data integrity story

Reply:
I beg to differ. Declarative API is working with declared arguments. Leaves us with no flexibility to use a query based on any variation, for example email, from date, to date, state, reference, etc.